packages/atomic/src/components/product-listing/atomic-product-listing/atomic-product-listing.tsx

```diff
@@ -0,0 +1,292 @@
+ *
+ * @part result-list - The element containing the list of results.
+ * @part result-list-grid-clickable-container - The parent of the result & the clickable link encompassing it.
+ * @part result-list-grid-clickable - The clickable link encompassing the result.
+ * @part label - The label of the result list.
+ * @part previous-button - The previous button.
+ * @part next-button - The next button.
+ * @part indicators - The list of indicators.
+ * @part indicator - A single indicator.
+ * @part active-indicator - The active indicator.
+ */
+@Component({
+  tag: 'atomic-product-listing',
+  styleUrl: 'atomic-product-listing.pcss',
+  shadow: true,
+})
+export class AtomicProductListing
+  implements InitializableComponent<ProductListingBindings>
+{
+  @InitializeBindings() public bindings!: ProductListingBindings;
+  public productListing!: ProductListing;
+  public resultsPerPage!: ResultsPerPage;
+  private resultListCommon!: ResultListCommon;
+  private loadingFlag = randomID('firstProductListingLoaded-');
+  private resultRenderingFunction: ResultRenderingFunction;
+
+  @Element() public host!: HTMLDivElement;
+
+  @State() public error!: Error;
+  @State() private resultTemplateRegistered = false;
+  @State() private templateHasError = false;
+  // @State() private currentPage = 0;
+  @BindStateToController('resultsPerPage')
+  @State()
+  public resultsPerPageState!: ResultsPerPageState;
+  @BindStateToController('productListing')
+  @State()
+  public productListingState!: ProductListingState;
+
+  @FocusTarget()
+  private nextNewResultTarget!: FocusTargetController;
+  /**
+   * The layout to apply when displaying results themselves. This does not affect the display of the surrounding list itself.
+   * To modify the number of products per column, modify the --atomic-product-listing-number-of-columns CSS variable.
+   */
+  @Prop({reflect: true}) public display: ResultDisplayBasicLayout = 'list';
+  /**
+   * The spacing of various elements in the result list, including the gap between results, the gap between parts of a result, and the font sizes of different parts in a result.
+   */
+  @Prop({reflect: true}) public density: ResultDisplayDensity = 'normal';
+  /**
+   * The expected size of the image displayed in the results.
+   */
+  @Prop({reflect: true})
+  public imageSize: ResultDisplayImageSize = 'small';
+
+  /**
+   * The total number of products to display.
+   * This does not modify the number of products per column. To do so, modify the --atomic-product-listing-number-of-columns CSS variable.
+   */
+  @Prop({reflect: true}) public numberOfProducts = 30;
+  /**
+   * The non-localized label for the list of products.
+   */
+  @Prop({reflect: true}) public label?: string;
+
+  /**
+   * The [heading level](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements) to use for the heading label, from 1 to 6.
+   */
+  @Prop({reflect: true}) public headingLevel = 0;
+  // @Watch('numberOfProductsPerPage')
```

Comment:
Is the intention to work on all the commented code in this PR in another PR ?

Reply:
Forgot to remove it sorry.